### PR TITLE
Removes static ctor from System.Math

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

**Comment by tannergooding** [comment]
The same change, if taken, should also happen for `System.MathF`

File: src/System.Private.CoreLib/shared/System/Math.cs [hunk 1]
```diff
@@ -26,16 +26,6 @@ public static partial class Math
 
         public const double PI = 3.14159265358979323846;
 
-        private const int maxRoundingDigits = 15;
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Math.cs]: Why is this static constructor a problem? It is before-field-init, so it should not cost anything unless you actually access the fields.

Comment (jkotas) [reply src/System.Private.CoreLib/shared/System/Math.cs]: We assume in many places that before-field-init static constructors costs nothing unless you actually access the fields.

File: src/System.Private.CoreLib/shared/System/Math.cs [hunk 2]
```diff
@@ -26,16 +26,6 @@ public static partial class Math
...
-
-        private static double doubleRoundLimit = 1e16d;
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Math.cs]: Can this be changed to `const`? It is a constant.

File: src/System.Private.CoreLib/shared/System/Math.cs (hunk 1 above)
Comment (marek-safar) [reply src/System.Private.CoreLib/shared/System/Math.cs]: Yeah, it's probably better to update our tooling to catch up with that.

File: src/System.Private.CoreLib/shared/System/Math.cs (hunk 2 above)
Comment (marek-safar) [reply src/System.Private.CoreLib/shared/System/Math.cs]: Already done that but will move this change to separate PR

**Pull request closed without merging**